### DISCUSSION TITLE
[QA] `--test` build

### DISCUSSION THREAD

**Discussion by vanessayuenn** [storybookjs/storybook#25695] [Announcement]
#### Feature owner: @ndelangen 
Please comment in this discussion thread for any bug report and feedback for this feature as you test out Storybook 8. Kindly adhere to the following format:

> **Project**: Name of your project as entered in Notion
> **Version**: The version of storybook you tested with
> **Bug/feedback**: Details of the bugs/feedback you have

Feature owner should triage the comments here frequently and open issues as needed.

**Comment by valentinpalkovic** [comment]
**Project**: Bitwarden Clients (Angular)
**Version**: 8.0.0-alpha.15
**Bug/feedback**: Test flag not recognized

The test flag isn't part of the start and build schema. Therefore, adding it as a builder option via angular.json is prohibited. Storybook's Angular builder also doesn't pass the flag further to Storybook's build and dev functions.

List of files that need adjustments:
- https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/builders/build-storybook/index.ts
- https://github.com/storybookjs/storybook/tree/next/code/frameworks/angular/src/builders/start-storybook/index.ts
- https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/builders/start-storybook/schema.json
- https://github.com/storybookjs/storybook/blob/next/code/frameworks/angular/src/builders/build-storybook/schema.json

**Reply by valentinpalkovic** [reply]
fixed

**Comment by ndelangen** [comment]
start-storybook shouldn't have it @valentinpalkovic 
`--test` is only for builds

**Comment by fredvisser** [comment]
**Project**: Nimble (html)
**Version**: 8.0.0-beta.2
**Bug/feedback**: 

In [our Storybook](https://nimble.ni.dev/storybook/?path=/docs/getting-started--docs), the `--test` flag appears to use a different minify strategy which breaks one of our more [complex/dynamic stories](https://60e89457a987cf003efc0a5b-usutpwboyy.chromatic.com/?path=/story/components-icons--icons). Noticed this when trying to use the [Chromatic Visual Test Addon](https://storybook.js.org/blog/chromatic-visual-test-addon-private-beta/?ref=storybookblog.ghost.io).

By using the --debug-webpack option, we see the configs being used, and here are some important differences:

![image](https://github.com/storybookjs/storybook/assets/1458528/c630bad5-06e2-4486-966e-6bb4c785ec90)

Namely, "our" build is using terserMinify with (notably) the options mangle: false and keep_fnames: true, whereas the addon's build is using esbuildMinify. If I add the following line to our .storybook/main.js in webpackFinal, the addon build works properly: config.optimization.minimizer[0].options.minimizer.options.keepNames = true; However, that is not a valid option for terserMinify, so it causes "our" build to barf.

Do we have the ability to directly configure the `--test` webpack config, or could it be a bit more conservative in the minify strategy by default?

**Reply by fredvisser** [reply]
Looks like the simplest fix was to be explicit about using the `terser-webpack-plugin` within our `.storybook/main.js` file, and then both the regular and `--test` builds use the same config.

I.e. 
```js
    const TerserPlugin = require('terser-webpack-plugin');
    config.optimization.minimizer = [
        new TerserPlugin({
            terserOptions: {
                keep_classnames: true,
                keep_fnames: true
            }
        })
    ];
```
